crates/pathfinder/src/core.rs

```diff
@@ -369,3 +369,71 @@ impl std::fmt::Display for Chain {
         }
     }
 }
+
+/// Adds a thin Debug and Disply implementations.
+///
+/// Debug impl produces `WrapperType(hash)` instead of automatic derives generated
+/// `WrapperType(StarkHash(hash))`.
+///
+/// Display impl produces just `hash`.
+macro_rules! thin_starkhash_debug_display {
+    ($target:ty) => {
+        thin_starkhash_debug!($target);
+
+        impl std::fmt::Display for $target {
+            fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+                std::fmt::Display::fmt(&self.0, f)
+            }
+        }
+    };
+
+    ($head:ty, $($tail:ty),+ $(,)?) => {
+        thin_starkhash_debug_display!($head);
+        thin_starkhash_debug_display!($($tail),+);
+    };
+}
+
+macro_rules! thin_starkhash_debug {
```

Comment:
```suggestion
macro_rules! thin_newtype_debug {
```

Reply:
I didn't choose this because we have other kinds of newtypes around H128 and u128 for example.